packages/expo-cli/src/commands/build/AndroidBuilder.js

```diff
@@ -19,16 +19,22 @@ export default class AndroidBuilder extends BaseBuilder {
     const sdkVersion = await this.validateProject(buildOptions);
     const { releaseChannel } = options;
     // Check the status of any current builds
-    await this.checkStatus({ platform: 'android', sdkVersion, releaseChannel, ...buildOptions });
+    await this.checkForBuildInProgress({
+      platform: 'android',
+      sdkVersion,
+      releaseChannel,
+      ...buildOptions,
+    });
     // Check for existing credentials, collect any missing credentials, and validate them
     await this.collectAndValidateCredentials(buildOptions);
     // Publish the current experience, if necessary
     let publishedExpIds = options.publicUrl ? undefined : await this.ensureReleaseExists('android');
 
+    await this.checkStatusBeforeBuild({ platform: 'android', sdkVersion, releaseChannel });
```

Comment:
We should skip calling this function if a user has provided the `--public-url` parameter.